### ISSUE TITLE
Random issue : test_adding_multiple_shapes_with_material_tag_to_reactor

### ISSUE THREAD

**Issue by RemDelaporteMathurin** [ukaea/paramak#409]
This test randomly fails. I think this is due to this line which seems to randomise the order of the `.material_tags` attribute.

https://github.com/ukaea/paramak/blob/23d50e9a81c1be846016dba106c8f6590308f78e/paramak/reactor.py#L76

In this minimal working example, `count` is sometimes 0, sometimes 500.
```python
count = 0
for i in range(500):
    a = list(set(['mat1', 'mat2']))
    if a[0] != "mat1":
        count += 1
print(count)
```

I don't understand why though...

**Comment by RemDelaporteMathurin** [comment]
https://stackoverflow.com/questions/9792664/converting-a-list-to-a-set-changes-element-order

**Comment by RemDelaporteMathurin** [comment]
@Shimwell is `set()` really needed here ?

**Comment by Shimwell** [comment]
Ah I see. So I was using ``set`` to avoid duplicate entries that are the same.

But I understand that set does not preserve the order.

Perhaps the use of ```set``` can be changed to something like this ```[x for x in a if x not in b]```

**Comment by Shimwell** [comment]
Or we can change the test to allow any order within the list

**Comment by RemDelaporteMathurin** [comment]
In #405 I removed the list(set()) transform since values is already a list

https://github.com/ukaea/paramak/blob/646cab369e1a19e6019414d0caa6f3caff53bada/paramak/reactor.py#L65-L76

**Comment by RemDelaporteMathurin** [comment]
Fixed in #421
